### PR TITLE
Javascript capstone project

### PR DESCRIPTION
**Pull request by LionRouge1**
In the accomplishment of this project we did the following tasks :
- We use webpack
- Install linters
- Use API to get data
- Implement the interface of the website
- Implement comment pop up
- Implement interface mobile version
- Set up a JavaScript testing library (Jest)

### CODE REVIEW

File: src/modules/Likes.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  async postLikes() {
+    const option = {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+      body: JSON.stringify(this.data),
+    };
```
Comment (nicupop729) [inline src/modules/Likes.js:31]: At the moment, a new like for the recipes is reflected only after clicked/liked, but on a refresh, it won't be reflected anymore. My like is lost :(

- [ ] Kindly check if the problem is that the like is not POSTed correctly to the Involvement API or that is not GET correctly. Narrow down the error by excluding possibilities of error one by one.

File: src/modules/comment.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      comment,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  commentList.innerHTML = '';
+  getComment(itemId);
+};
+
```
Comment (nicupop729) [inline src/modules/comment.js:68]: As for the likes, the same error occurs here, for the comments. A comment won't persist after a refresh. 

- [ ] Kindly check if is posted correctly or if you get the comments correctly from the API.

Hint: you can make use of the [Postman services](https://www.postman.com) or [Insomnia](https://insomnia.rest) to check if your function works correctly and that you post to the API.

File: dist/index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            <div>
+            <ul class ="comment-list">
+                    
+            </ul>
+        </div>
+            <form class="comment">
+                <h2 class ="add-comment">Add a comment</h2>
+                 <input type="text" name="name" id="name" placeholder="your name" class="comment-input" required><br>
+                 <textarea id="txtid" name="txtname" class="comment-area" rows="3" cols ="20" required>Write a comment</textarea><br>
+                <button class="comment-place-btn" type="submit">Comment</button>
```
Comment (nicupop729) [inline dist/index.html:50]: When the Comment button is clicked, the user is redirected to the home page. What should happen is that the comment should be displayed on the UI and the data should be recorded by the Involvement API.

- [ ] Please, fix this issue, in order that after the button is clicked, the user remains on that particular recipe, with it's comment recorded and updated on the UI. :+1:

**Review by nicupop729 — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️

Hi @LionRouge1 & @IfzaRasool, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### Some highlights for you
* Correct gitflow practice ✔️ 
* Good commit messages ✔️ 
* Linters checks pass flawlessly :heavy_check_mark: 

### Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+const container = document.querySelector('.meals_container');
+
+const display = async () => {
+  const meals = await getData('https://www.themealdb.com/api/json/v1/1/search.php?f=a');
+  let liked = [];
+  if (localStorage.getItem('liked') != null) {
+    const lik = JSON.parse(localStorage.getItem('liked'));
+    liked = lik.map((item) => item.item_id);
+  }
+
```
Comment (Gambit142) [inline src/index.js:25]: - [ ] I noticed that when I click the comment button and complete a comment, everything works properly but I noticed that if I exit the popup and try to click the comment button, I get a blank screen until I refresh the page before I see contents. Kindly fix it and make your application work fine. I recorded a gif to explain the bug in the application.

chrome-capture (2)

**Review by Gambit142 — APPROVED** [review]
Hi @LionRouge1 @IfzaRasool 👋, 

Great work team implementing all the changes that the former previewer suggested. 👍 💯 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

#### Kindly consider the optional suggestion I made under the review and remove all unnecessary codes that are commented out in the `index.js` file. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 4 above)
Comment (IfzaRasool) [reply src/index.js:25]: Thank you 👍 I resolve it

**Pull request merged**